common/static/js/src/ReactRenderer.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class ReactRendererException extends Error {
+  constructor (...args) {
+    super(...args);
+    this.name = 'ReactRendererException';
+    Error.captureStackTrace(this, ReactRendererException);
+  }
+  toString() {
+    return `${this.name}: ${this.message}`;
+  }
+}
+
+export class ReactRenderer {
+  constructor({component, selector, componentName, props}) {
+    Object.assign(this, {
+      component,
+      selector,
+      componentName,
+      props,
+    });
+    this.handleArgumentErrors();
+    this.targetElement = this.getTargetElement(selector);
+    this.renderComponent();
+  }
+
+  handleArgumentErrors() {
+    if (this.component === null) {
+      throw new ReactRendererException(
+        `Component ${this.componentName} is not defined. Make sure you're ` +
+        `using a non-default export statement for the ${this.componentName} ` +
+        `class, that ${this.componentName} has an entry point defined ` +
+        `within the 'entry' section of webpack.common.config.js, and that the ` +
+        `entry point is pointing at the correct file path.`
+      );
+    }
+    if (this.props && !((!!this.props) && (this.props.constructor === Object))) {
```

Comment:
I could totally be misunderstanding this boolean so I apologize in advance...

It seems to me that we want to throw any time `this.props` is not an `Object`.

If `this.props` is `undefined` then `!!this.props` is `false` so the second `boolean` statement is `true` but since `undefined` is falsey the entire statement would evaluate to `false`.
```javascript
if (undefined && !(!!undefined && false))
if (undefined && !(false && false))
if (undefined && true)
if (false)
```

Also, if `this.props` is `false` then the entire statement boils down to
```javascript
if (false && !(!!false && false))
if (false && !(false))
if (false && true)
if (false)
```
which means that we wouldn't throw if `this.props` somehow was `undefined` or `false`.

I could totally be misunderstanding the intention of this check, but maybe 
```javascript
if (!(this.props instanceof Object && this.props.constructor === Object)
```
might be clearer ([SO article on checking if something is an `Object`](https://stackoverflow.com/a/46663081/5225575))?